backup: TestBackupJobRetryReset failed

backup.TestBackupJobRetryReset [failed](https://mesolite.cluster.engflow.com/invocations/default/40b885c5-848f-41bb-a590-65f73a3f415c?testReportRun=5&testReportShard=40&testReportAttempt=1#targets-Ly9wa2cvYmFja3VwOmJhY2t1cF90ZXN0) on master @ [9fc2f395971bd0b3a3336570639d3547b7ac479e](https://github.com/cockroachdb/cockroach/commits/9fc2f395971bd0b3a3336570639d3547b7ac479e):


```
=== RUN   TestBackupJobRetryReset
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestBackupJobRetryReset3749690308
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:638: DRPC is enabled (via testserver factory defaults)
    testutils.go:368: no Invalid Descriptors
    backup_test.go:1756: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupJobRetryReset3749690308
--- FAIL: TestBackupJobRetryReset (8.88s)
=== RUN   TestBackupJobRetryReset/fail_after_all_nodes_have_reported_progress
    backup_test.go:1753: 
        	Error Trace:	pkg/backup/backup_test.go:1753
        	Error:      	Not equal: 
        	            	expected: 12
        	            	actual  : 4
        	Test:       	TestBackupJobRetryReset/fail_after_all_nodes_have_reported_progress
        	Messages:   	unexpected number of attempts
--- FAIL: TestBackupJobRetryReset/fail_after_all_nodes_have_reported_progress (5.15s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=5</code>
 - <code>shard=40</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupJobRetryReset.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-61905

--- a/pkg/backup/backup_test.go
+++ b/pkg/backup/backup_test.go
@@ -1711,6 +1711,9 @@ func TestBackupJobRetryReset(t *testing.T) {
 	}
 	params.ServerArgs = base.TestServerArgs{Knobs: knobs}
 	_, sqlDB, _, cleanupFn := backupRestoreTestSetupWithParams(t, singleNode, 10, InitManualReplication, params)
+	// We disable range-sized spans for this test to avoid 0-span progress
+	// updates. The conditions in which this can occur are outlined in #166401.
+	sqlDB.Exec(t, "SET CLUSTER SETTING bulkio.backup.presplit_request_spans.enabled = false")
 	// This creates 4 additional backup chunks.
 	for i := 0; i < 4; i++ {
 		sqlDB.Exec(t, fmt.Sprintf(`CREATE TABLE %s AS (SELECT * FROM bank)`, "bank"+strconv.Itoa(i)))

